---
title: "Indicator Analysis Take 2"
author: "Alex Linz"
date: "Tuesday, November 17, 2015"
output: html_document
---

Problems addressed in this version of the indicator analysis:
- All phylogenetic levels are compared at once
- Unclassified groups removed at each level
- Abundance threshold specified
- # of permutations increased
- Correlation coefficients used instead of p-values as metric of "best" indicators

###Make a table that includes EVERY taxonomic level at once. 
So for example, it would include as separate rows:

k__Bacteria(100);p__Actinobacteria(100);c__Actinobacteria(100);o__Actinomycetales(100);acI(100);acI-A(70);acI-A7(60);
k__Bacteria(100);p__Actinobacteria(100);c__Actinobacteria(100);o__Actinomycetales(100);acI(100);acI-A(70);
k__Bacteria(100);p__Actinobacteria(100);c__Actinobacteria(100);o__Actinomycetales(100);acI(100);

This will allow me to choose the level that is the best indicator.

```{r, echo=TRUE}
library(OTUtable)
library(indicspecies)

data(otu_table)
data(taxonomy)

tribe_table <- combine_otus("Tribe", otu_table, taxonomy)
clade_table <- combine_otus("Clade", otu_table, taxonomy)
lineage_table <- combine_otus("Lineage", otu_table, taxonomy)
order_table <- combine_otus("Order", otu_table, taxonomy)
class_table <- combine_otus("Class", otu_table, taxonomy)
phylum_table <- combine_otus("Phylum", otu_table, taxonomy)

full_table <- rbind(otu_table, tribe_table, clade_table, lineage_table, order_table, class_table, phylum_table)
```

Remove unclassified groups from the table and specify an abundance threshold

```{r, echo=TRUE}
classified <- grep("unclassified", rownames(full_table))
classified1 <- grep("__$", rownames(full_table))
input_table <- full_table[-c(classified, classified1),]


```

The abundance threshold decreased the number of groups from 8000 to 2000. This should speed up the process.

Now run the indicator analysis, with 10,000 permutations

```{r, echo=TRUE}
seven <- year_subset("07", input_table)
#Abundance threshold is 75th percentile or greater. In this case, the number is 134 reads, which is not very strict at all.
threshold <- quantile(rowSums(seven))[4]
seven <- seven[which(rowSums(seven) >= threshold),]


seven <- t(seven)
seven <- as.data.frame(seven)

sampleids <- rownames(seven)
layer <- substr(sampleids, start=3, stop=3)
layerid <- c("E", "H")

epi_v_hypo <- c()
for(i in 1:length(layerid)){
  epi_v_hypo[which(layer == layerid[i])] <- i
}

#Run multipatt(), specifing the the index of choice is group-normalized correlation
clade_by_layer <- multipatt(x = seven, cluster = epi_v_hypo, func = "r.g", control = how(nperm = 9999))
write.csv(clade_by_layer$sign, file="Clade_by_layer_all_levels2007.csv")

###
nine <- year_subset("09", input_table)
threshold <- quantile(rowSums(nine))[4]
nine <- nine[which(rowSums(nine) >= threshold),]
nine <- t(nine)
nine <- as.data.frame(nine)

sampleids2 <- rownames(nine)
layer2 <- substr(sampleids2, start=3, stop=3)
layerid2 <- c("E", "H")

epi_v_hypo09 <- c()
for(i in 1:length(layerid2)){
  epi_v_hypo09[which(layer2 == layerid2[i])] <- i
}

clade_by_layer2 <- multipatt(x = nine, cluster = epi_v_hypo09, func = "r.g", control = how(nperm = 9999))
write.csv(clade_by_layer2$sign, file="Clade_by_layer_all_levels2009.csv")
```

Now repeat on mixing regime indicator analysis

```{r, echo=TRUE}
seven_hypo <- seven[which(layer == "H"),]
nine_hypo <- nine[which(layer2 == "H"),]


lakeid <- c("CB", "FB", "WS", "NS", "TB", "SS", "HK","MA")

lakes07 <- lakes <- substr(rownames(seven_hypo), start=1, stop=2)
lakes09 <- lakes <- substr(rownames(nine_hypo), start=1, stop=2)

mixing_groups07 <- c()
mixing_groups07[which(lakes07 == "CB" | lakes07 == "FB" | lakes07 == "WS")] <- 1
mixing_groups07[which(lakes07 == "TB"| lakes07 == "NS"| lakes07 == "SS")] <- 2
mixing_groups07[which(lakes07 == "MA" | lakes07 == "HK")] <- 3

mixing_groups09 <- c()
mixing_groups09[which(lakes09 == "CB")] <- 1
mixing_groups09[which(lakes09 == "TB"| lakes09 == "NS"| lakes09 == "SS")] <- 2
mixing_groups09[which(lakes09 == "MA")] <- 3

clades_by_mixing07 <- multipatt(x = seven_hypo, cluster = mixing_groups07, func = "r.g", control = how(nperm = 9999))

clades_by_mixing09 <- multipatt(x = nine_hypo, cluster = mixing_groups09, func = "r.g", control = how(nperm = 9999))

write.csv(clades_by_mixing07$sign, file = "Clades_by_mixing2007.csv")
write.csv(clades_by_mixing09$sign, file = "Clades_by_mixing2009.csv")