\nonstopmode{}
\documentclass[a4paper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8,latin1]{inputenc}
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge Package `OTUtable'}}
\par\bigskip{\large \today}
\end{center}
\begin{description}
\raggedright{}
\item[Type]\AsIs{Package}
\item[Title]\AsIs{North Temperate Lakes - Microbial Observatory 16S Time Series
Data and Functions}
\item[Version]\AsIs{1.1.0}
\item[Date]\AsIs{2017-04-12}
\item[Author]\AsIs{Alexandra Linz}
\item[Maintainer]\AsIs{Alexandra Linz }\email{amlinz16@gmail.com}\AsIs{}
\item[Description]\AsIs{Analyses of OTU tables produced by 16S sequencing, as well as example data. It contains the data and scripts used in the paper Linz, et al. (2017) ``Bacterial community composition and dynamics spanning five years in freshwater bog lakes'' (Manuscript submitted, preprint available at <http://biorxiv.org/content/early/2017/04/12/127035>).}
\item[License]\AsIs{GPL-3}
\item[LazyLoad]\AsIs{yes}
\item[Depends]\AsIs{R (>= 2.10)}
\item[NeedsCompilation]\AsIs{no}
\end{description}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{OTUtable-package}{OTU table analysis functions}{OTUtable.Rdash.package}
\aliasA{OTUtable}{OTUtable-package}{OTUtable}
\keyword{package}{OTUtable-package}
%
\begin{Description}\relax
Contains functions for the analysis of an OTU table generated from 16S rRNA amplicon sequencing. It also includes the data from the North Temperate Lakes-Microbial Observatory used in the paper Linz, et al. (2017) "Bacterial community composition and dynamics spanning five years in freshwater bog lakes" (Manuscript submitted, preprint available at <http://biorxiv.org/content/early/2017/04/12/127035>). The data and code used in this paper are available at <https://github.com/McMahonLab/North\_Temperate\_Lakes-Microbial\_Observatory>
Three data files are included: otu\_table, taxonomy, and metadata. Access these by calling them with data(). There is also a fasta file associated with this dataset that is not included in this package - it can be found on our GitHub page in Data/16S\_data. This package does not include functionality for fasta files; if you need this for analyses such as calculating UniFrac distance, please see the R package "phyloseq".
\end{Description}
%
\begin{Details}\relax

\Tabular{ll}{
Package: & OTUtable\\{}
Type: & Package\\{}
Version: & 1.1.0\\{}
Date: & 2017-04-11\\{}
License: & GPL-3 \\{}
}
Functions include:
bog\_subset \\{}
chao1 \\{}
clean\_shared \\{}
clean\_mothur\_taxonomy \\{}
clean\_TaxAss\_taxonomy \\{}
combine\_otus \\{}
extract\_date \\{}
filter\_taxa \\{}
grab\_group \\{}
make\_do\_matrix \\{}
make\_temp\_matrix \\{}
obs\_richness \\{}
pielou \\{}
plot\_column \\{}
reduce\_names \\{}
remove\_reps \\{}
rotate \\{}
shannon \\{}
strat\_metric \\{}
year\_subset \\{}
zscore \\{}
\end{Details}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{References}\relax
Linz, A.M., Crary, B.C., Shade, A., Owens, S., Gilbert, J.A., Knight, R., and McMahon, K.D. (2017) "Bacterial community composition and dynamics spanning five years in freshwater bog lakes" (Manuscript submitted, preprint available at <http://biorxiv.org/content/early/2017/04/12/127035>)
\end{References}
\inputencoding{utf8}
\HeaderA{bog\_subset}{Subset OTU table by sampling site}{bog.Rul.subset}
\keyword{data\_processing}{bog\_subset}
%
\begin{Description}\relax
Returns an OTU table containing only samples from the identified sampling site. This function can also be used on tables of higher level taxa generated by combine\_otus(), or on tables that have already been processed by year\_subset().
\end{Description}
%
\begin{Usage}
\begin{verbatim}
bog_subset(bog_id, table)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{bog\_id}] 
The three letter code indicating the sampling site. The bog is represented by letters one and two; options are TB, SS, CB, NS, MA, HK, WS, and FB. The third letter indicates the layer; E for epilimnion and H for hypolimnion. The bog\_id should be in quotes, and regular expressions can be used.

\item[\code{table}] 
A table containing the relative abundances of each taxon as rows and samples as columns. Sample names must be coded in the format bog, layer, date, and replicate (example: TBE07JUN08.R2 == Trout Bog Epilimnion, collected 07Jun08, replicate 2)

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a relative abundance table containing samples from the specified sampling site in columns, with taxa in rows
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(otu_table)

Trout_Bog_Epilimnion <- bog_subset("TBE", otu_table)

Hells_Kitchen_Hypolimnion <- bog_subset("HKH", otu_table)

# Include both epilimnion and hypolimnion in a single table
Trout_Bog_both_layers <- bog_subset("TB.", otu_table)

# Include all meromictic hypolimnia
meromictic_hypolimnia <- bog_subset("HKH|MAH", otu_table)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{chao1}{Chao1 Richness}{chao1}
\keyword{biodiversity}{chao1}
%
\begin{Description}\relax
Calculates Chao1 richness of a vector of relative abundance data. This alpha diversity metric takes into account the number of singletons and doubletons for a more accurate estimate than observed richness.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
chao1(sample)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sample}] 
A vector of relative abundance data, typically a column in a matrix

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a single number indicating the estimated richness in the tested sample based on the number of taxa appearing only once or twice
\end{Value}
%
\begin{Note}\relax
Use apply functions to calculate Chao1 richness for all samples in a matrix

\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(otu_table)
chao1_richness <- apply(otu_table, 2, chao1)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{clean\_mothur\_taxonomy}{Clean mothur-format Taxonomy File}{clean.Rul.mothur.Rul.taxonomy}
\keyword{format\_mothur\_output}{clean\_mothur\_taxonomy}
%
\begin{Description}\relax
Reduces information in a mothur .taxonomy file by removing the second column with the number of reads per OTU. It will also check for and remove OTUs in the taxonomy file that are not in OTU table - this may be the case if rarefaction was performed after classification, as was used in the NTL-Microbial Observatory dataset. This function was formerly clean\_taxonomy in v1.0.0.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
clean_mothur_taxonomy(taxonomy_file, table, remove_bootstrap)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{taxonomy\_file}] 
A .taxonomy file output by mothur

\item[\code{table}] 
An OTU table containing OTU numbers as row names

\item[\code{remove\_bootstrap}] 
TRUE or FALSE: if TRUE, removes bootstrap values from the classification strings

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns the taxonomy with OTUs as row names and seven columns containing each taxonomic level (Kingdom, Phylum, Class, Order, Lineage, Clade, and Tribe)
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

# Example path only: path <- "mothur_output/bogs.taxonomy"
# table <- clean_shared("mothur_output/bogs.shared", trim.names = T)
# taxonomy <- clean_mothur_taxonomy(path, table, remove_bootstrap = F)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{clean\_shared}{Reformat a shared file}{clean.Rul.shared}
\keyword{format\_mothur\_output}{clean\_shared}
%
\begin{Description}\relax
Converts a mothur .shared file into a simplified OTU table. The columns indicating total reads for each OTU and the clustering level are removed, and the table is transposed so that OTUs are rows and samples are columns. The "trim.names" variable provides an option to shorten sample names to the first "." character - this is specific to the NTL-Microbial Observatory dataset. Manual curation of sample names took place after this step for the NTL-Microbial Observatory dataset in order to maintain consistency across all sample names.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
clean_shared(shared_file, trim.names)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{shared\_file}] 
A .shared file output by mothur

\item[\code{trim.names}] 
TRUE or FALSE - if TRUE, sample names will be trimmed to the first "." character.

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns an OTU table with samples as columns and OTUs as rows. 
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

# Example path only: path <- "mothur_output/bogs.shared"

# otu_table <- clean_shared(path, trim.names = T)
# write.csv(otu_table, file = "bogs_otu_table.csv", quote = F, row.names = T)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{clean\_TaxAss\_taxonomy}{Clean Taxonomy File Output by TaxAss Workflow}{clean.Rul.TaxAss.Rul.taxonomy}
\keyword{format\_mothur\_output}{clean\_TaxAss\_taxonomy}
%
\begin{Description}\relax
Formats a taxonomy file output by the McMahon Lab TaxAss 16S classification workflow (github.com/McMahonLab/TaxAss) into the same format produced by clean\_mothur\_taxonomy(). It will also check for and remove OTUs in the taxonomy file that are not in OTU table - this may be the case if rarefaction was performed after classification, as was used in the NTL-Microbial Observatory dataset.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
clean_TaxAss_taxonomy(taxonomy_file, table, remove_bootstrap)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{taxonomy\_file}] 
A .taxonomy file output by the TaxAss workflow

\item[\code{table}] 
An OTU table containing OTU numbers as row names

\item[\code{remove\_bootstrap}] 
TRUE or FALSE: if TRUE, removes bootstrap values from the classification strings

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns the taxonomy with OTUs as row names and seven columns containing each taxonomic level (Kingdom, Phylum, Class, Order, Lineage, Clade, and Tribe)
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

# Example path only: path <- "TaxAss_output/bogs.taxonomy"
# table <- clean_shared("mothur_output/bogs.shared", trim.names = T)
# taxonomy <- clean_TaxAss_taxonomy(path, table, remove_bootstrap = F)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{combine\_otus}{Combine OTUs based on identical taxonomic assignments}{combine.Rul.otus}
\keyword{data\_processing}{combine\_otus}
%
\begin{Description}\relax
Sums the abundances of OTUs with the same taxonomy at a given level into a single vector for that taxonomy. This creates a new table of relative abundance data at a higher taxonomic level than OTU. This function only works with the OTU level as input, but can be used on any subset of the OTU table created by year\_subset() or bog\_subset(). The OTU table must have the same number of rows as the taxonomy file (do not remove rows with no reads before running combine\_otus())
If bootstrap values were not removed by expand\_taxa(), this command will likely create spurious groupings based on identical bootstrap values.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
combine_otus(level, table, taxonomy)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{level}] 
The desired level at which to combine OTUs; options are the column names from the taxonomy dataset

\item[\code{table}] 
An OTU table containing the relative abundances of each OTU.

\item[\code{taxonomy}] 
A taxonomy dataset in the form produced by expand\_taxa().

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a table of relative abundance data with each row representing all OTUs of a given taxonomic assignment summed together. Row names are now the full taxonomic assignment of each row. To keep only the the lowest taxonomic level in the row names, run the function reduce\_names()
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

data(otu_table)
data(taxonomy)

example_table <- year_subset("05", otu_table)
example_table <- bog_subset("TBE", example_table)

phylum_table <- combine_otus("Phylum", example_table, taxonomy)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{extract\_date}{Extract sampling date from a vector of sample names}{extract.Rul.date}
\keyword{data\_processing}{extract\_date}
%
\begin{Description}\relax
The date each sample was collected is encoded in the sample ID. Extract this into R date format using this command. 
\end{Description}
%
\begin{Usage}
\begin{verbatim}
extract_date(sample_ids)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sample\_ids}] 
A vector of sample names. Samples must be labeled using the bog, layer, date, and replicate system (MAH04JUL05.R1 = Mary Lake Hypolimnion, 04Jul05, replicate 1)

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a vector of dates corresponding to each sample
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

samples <- c("TBE01JUN09.R1", "TBE05JUN09", "TBE10JUN09.R2")
extract_date(samples)

# Extract sample dates from the OTU table
data(otu_table)
x <- extract_date(colnames(otu_table))

# Extract sample dates from the metadata
data(metadata)
x <- extract_date(metadata$Sample_Name)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{filter\_taxa}{Filter Taxa Based on Abundance and Persistence}{filter.Rul.taxa}
\keyword{data\_processing}{filter\_taxa}
%
\begin{Description}\relax
Returns a table containing only taxa that meet the imposed requirements of a minimum abundance and a minimum number of samples containing that taxon
\end{Description}
%
\begin{Usage}
\begin{verbatim}
filter_taxa(table, abundance, persistence)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{table}] 
A table containing the relative abundances of each OTU or taxon in the form produced by clean\_shared(). Can be used on the output of grab\_groups() or combine\_otus()

\item[\code{abundance}] 
The minimum threshold for percentage of reads attributed to a taxon in at least one sample. Taxa at abundances greater than or equal this number will be retained.

\item[\code{persistence}] 
The minimum threshold for the percentage of samples in which a taxon has been observed.  Taxa at abundances greater than or equal this number will be retained.

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a table with all taxa that met the imposed thresholds
\end{Value}
%
\begin{Note}\relax
Thanks Juliana Dias for suggesting this function!

\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

# To make a table containing only OTUs with at least 0.1% abundance 
# in at least one sample that were observed 
# (at any abundance) in at least 50% of samples:
# library(OTUtable)
# data(otu_table)
# filtered_table <- filter_taxa(otu_table, abundance = 0.1, persistence = 50)

# To make a table containing only phyla with at least 10% abundance 
# in any one sample and were observed 
# at any abundance in at least 10% of samples:
# data(taxonomy)
# phylum_table <- combine_otus("Phylum", otu_table, taxonomy)
# filtered_phylum_table <- filter_taxa(phylum_table, abundance = 10, persistence = 10)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{grab\_group}{Subset OTU table by taxonomic assignment}{grab.Rul.group}
\keyword{data\_processing}{grab\_group}
%
\begin{Description}\relax
Returns a table containing only taxa from a given phylogenetic group
\end{Description}
%
\begin{Usage}
\begin{verbatim}
grab_group(group, level, table, taxonomy)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{group}] 
The phylogenetic classification of interest (can be a regular expression)

\item[\code{level}] 
The phylogenetic level of the group of interest (must be a column name in the taxonomy file)

\item[\code{table}] 
A table containing the relative abundances of each OTU in the form produced by clean\_shared()

\item[\code{taxonomy}] 
A taxonomy dataset in the form produced by expand\_taxa()

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a table with all taxa of a given taxonomic assignment
\end{Value}
%
\begin{Note}\relax
This function must be run on the OTU level table. However, the output of this function can be run through combine\_otus() to create a higher level table of results. Sometimes closely related groups were classified better in the Greengenes vs the freshwater database during classification of the NTL-Microbial Observatory dataset. In this case, it is necessary to search for the names generated by both datasets to get all closely related OTUs.
For example, Methylophilaceae in Greengenes are named betIV in the freshwater database.

\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
 
data(otu_table)
data(taxonomy)

acI <- grab_group("acI", "Clade", otu_table, taxonomy)
verruco <- grab_group("Verrucomicrobia", "Phylum", otu_table, taxonomy)

# Example where two search terms are needed due to classification with two databases
methylophilaceae <- grab_group("Methylophilaceae|betIV", "Clade", otu_table, taxonomy)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{make\_do\_matrix}{Make matrix of dissolved oxygen data}{make.Rul.do.Rul.matrix}
\keyword{water\_column\_plots}{make\_do\_matrix}
%
\begin{Description}\relax
Takes a given sample ID and converts the dissolved oxygen data in data(metadata) from long format into a matrix. This is useful for plotting using plot\_column()
\end{Description}
%
\begin{Usage}
\begin{verbatim}
make_do_matrix(sampleID, field_data)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sampleID}] 
A regular expression used to select a group of samples

\item[\code{field\_data}] 
A dataset of DO profiles in long format. Column names must be the same as the metadata file provided with this package

\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Also fills in NA values with the average of the depth above and below the missing value. If the value is at the bottom of the water column, the second deepest is substituted.
\end{Details}
%
\begin{Value}
Returns matrix of DO data with depth in rows and date in columns
\end{Value}
%
\begin{Note}\relax
This is mainly used for generating contour plots. In general, long format is easier to work with.
In the metadata file included in this package, each DO measurement is listed twice, once under the epilimnion sample name and again under the hypolimnion sample name.
\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(metadata)

dissolved_oxygen <- make_do_matrix("TBE.....07", metadata)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{make\_temp\_matrix}{Make matrix of temperature data}{make.Rul.temp.Rul.matrix}
\keyword{water\_column\_plots}{make\_temp\_matrix}
%
\begin{Description}\relax
Takes a given sample ID and converts temperature data of water profiles over time from long format into a matrix. This is most often useful for plotting using plot\_column().
\end{Description}
%
\begin{Usage}
\begin{verbatim}
make_temp_matrix(sampleID, field_data)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sampleID}] 
A regular expression used to select a group of samples

\item[\code{field\_data}] 
A dataset of temperature profiles in long format. Column names must be the same as the metadata file provided with this package

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns matrix of temperature data with depth in rows and date in columns
\end{Value}
%
\begin{Note}\relax
This is mainly used for generating contour plots. In general, long format is easier to work with.
In the included metadata file, each temperature measurement is recorded twice, once as epilimnion and once as hypolimnion.
\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(metadata)

temp <- make_temp_matrix("TBE.....07", metadata)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{metadata}{Lake metadata for OTU table}{metadata}
%
\begin{Description}\relax
A dataset containing temperature and oxygen profiles from the lakes in this study

\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(taxonomy)
\end{verbatim}
\end{Usage}
%
\begin{Format}
A dataframe with 6 columns (measured variables) and 13,607 rows	(depth profiles)
\end{Format}
%
\begin{Details}\relax
Missing data indicated by NA
Some sample dates and metadata dates may not match up exactly; if this presents an issue, please email and I will look at our written records for the right date
Epilimnion and hypolimnion samples each have an identical depth profile entry; search for just one or the other
\end{Details}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
\inputencoding{utf8}
\HeaderA{obs\_richness}{Observed Richness}{obs.Rul.richness}
\keyword{biodiversity}{obs\_richness}
%
\begin{Description}\relax
Calculates observed richness on a single column of relative abundance data. 
\end{Description}
%
\begin{Usage}
\begin{verbatim}
obs_richness(sample)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sample}] 
A vector of relative abundance data, typically a single column in a matrix

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a single number indicating the number of taxa in the tested sample
\end{Value}
%
\begin{Note}\relax
Use apply functions to calculate richness for all samples in a matrix

\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
 data(otu_table)
 richness <- apply(otu_table, 2, obs_richness)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{otu\_table}{OTU table generated from 8 bog lakes over 4 years}{otu.Rul.table}
%
\begin{Description}\relax
A dataset containing bacterial relative abundance data from the North Temperate Lakes Microbial Observatory
Produced from mothur output using clean\_shared()
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(otu_table)
\end{verbatim}
\end{Usage}
%
\begin{Format}
A dataframe with 1,387 columns (samples) and 6,208 rows	(OTUs)
\end{Format}
%
\begin{Details}\relax
Contains replicate samples
Each column has been rarefied to 2500
Sample names encode sampling site ("TB"), epilimnion or hypolimnion ("E" or "H"), sampling date ("01JUN07") and replicate(".R2")
\end{Details}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
\inputencoding{utf8}
\HeaderA{pielou}{Pielou's Evenness}{pielou}
\keyword{biodiversity}{pielou}
%
\begin{Description}\relax
Calculates Pielou's evenness for a single vector of relative abundance data
\end{Description}
%
\begin{Usage}
\begin{verbatim}
pielou(sample)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sample}] 
A vector of relative abundance data

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a single value indicating the evenness of a community
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(otu_table)
even <- apply(otu_table, 2, pielou)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{plot\_column}{Plot DO or temperature data from a depth profile over time}{plot.Rul.column}
\keyword{water\_column\_plots}{plot\_column}
%
\begin{Description}\relax
Takes output from make\_do\_matrix or make\_temp\_matrix and plots using filled.contour()
\end{Description}
%
\begin{Usage}
\begin{verbatim}
plot_column(data_matrix, title)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{data\_matrix}] 
A matrix output by make\_do\_matrix() or make\_temp\_matrix()

\item[\code{title}] 
The title you would like on the plot

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Plots a filled contour plot showing the water column over time
\end{Value}
%
\begin{Note}\relax
Depends on the function rotate(). The functions make\_do\_matrix() and make\_temp\_matrix() fill in missing values with the average of the measurement at each depth above and below; however, if missing values are present in the matrix for plotting, these will appear as white space on the plot.
\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(metadata)
temp <- make_temp_matrix("TBE.....07", metadata)
plot_column(temp, "Trout Bog 2007 Temperature")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{reduce\_names}{Shorten taxonomic assignment in table row names}{reduce.Rul.names}
\keyword{data\_processing}{reduce\_names}
%
\begin{Description}\relax
Reduces the full string indicating taxonomy to the last classified level. Works on tables at levels higher than OTUs.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
reduce_names(table)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{table}] 
A table containing the relative abundances of each taxa produced by combine\_otus()

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns the same table with shortened row names
\end{Value}
%
\begin{Note}\relax
This function is often most useful for plotting, so that the full string does not appear on the plot
\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(otu_table)
data(taxonomy)

# Create a small table for the example
# example <- year_subset("05", otu_table)
# example <- bog_subset("TBE", example)

# clade_table <- combine_otus("Clade", example, taxonomy)
# clade_table <- clade_table[which(rowSums(clade_table) > 0),]
# head(rownames(clade_table))
# reduced_clades <- reduce_names(clade_table)
# head(rownames(reduced_clades))
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{remove\_reps}{Remove the second replicate of each sample, when it exists}{remove.Rul.reps}
\keyword{data\_processing}{remove\_reps}
%
\begin{Description}\relax
Sometimes it is desirable to remove replicate samples (often for plotting). This command removes all samples marked as replicate 2. Please note that you should always check the similarity of replicates for your metric of interest before removing them for aesthetic purposes.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
remove_reps(table)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{table}] 
An OTU table containing the relative abundances of each OTU

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns an OTU table containing only one replicate for each sample
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
 data(otu_table)
 no_reps <- remove_reps(otu_table)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{rotate}{Rotate a matrix}{rotate}
\keyword{water\_column\_plots}{rotate}
%
\begin{Description}\relax
Rotates a matrix of data so that columns are reversed
\end{Description}
%
\begin{Usage}
\begin{verbatim}
rotate(data_matrix)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{data\_matrix}] 
Used in this package with matrix output by make\_do\_matrix or make\_temp\_matrix as part of the function plot\_column(). Any matrix will work, though.


\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Used to rotate the DO or temperature matrices so that depth 0 is at the top of a contour plot and the max depth is at the bottom.
\end{Details}
%
\begin{Value}
Returns a matrix that has been rotated so that it reads from bottom to top
\end{Value}
%
\begin{Note}\relax
Used with make\_do\_matrix(), make\_temp\_matrix(), and plot\_column(). plot\_column() depends on this function.
\end{Note}
%
\begin{Author}\relax
An anonymous author on Stack Overflow
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(metadata)
temp <- make_temp_matrix("TBE.....07", metadata)
r_temp <- rotate(temp)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{shannon}{Shannon's Biodiversity Index}{shannon}
\keyword{biodiversity}{shannon}
%
\begin{Description}\relax
Calculates Shannon's Biodiversity Index on a single column of relative abundance data. This metric takes into account both richness and evenness.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
shannon(sample)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sample}] 
A vector of relative abundance data, typically a single column in a matrix

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a single number indicating the amount of biodiversity in the tested sample
\end{Value}
%
\begin{Note}\relax
Use apply functions to calculate Shannon's index for all samples in a matrix

\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(otu_table)
richness <- apply(otu_table, 2, shannon)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{taxonomy}{Taxonomic assignments of OTUs}{taxonomy}
%
\begin{Description}\relax
A dataset containing the taxonomy of each OTU in the otu\_table
Produced from mothur output using clean\_taxonomy()
Bootstrap values have been removed from this dataset, but are still available in as part of the Data folder in the McMahonLab/North\_Temperate\_Lakes-Microbial\_Observatory GitHub repo
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(taxonomy)
\end{verbatim}
\end{Usage}
%
\begin{Format}
A dataframe with 7 columns (taxonomic levels) and 6,208 rows	(OTUs)
\end{Format}
%
\begin{Details}\relax
Classified using our Freshwater database, followed by Greengenes - for the full workflow, visit the McMahonLab Github 16STax-Ass repository
Some OTUs are missing; these were removed by subsampling of the OTU table
The presence of both blank (\_\_) assignments and "unclassified" assignments are the result of the dual classification.
\end{Details}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
\inputencoding{utf8}
\HeaderA{year\_subset}{Subset samples by a specific year}{year.Rul.subset}
\keyword{data\_processing}{year\_subset}
%
\begin{Description}\relax
Takes the year value in the last two digits of the sample ID and allows selection of a single year of data. Can be performed on tables at higher taxonomic levels generated by combine\_otus(), or on tables already subset by bog\_subset().
\end{Description}
%
\begin{Usage}
\begin{verbatim}
year_subset(year_id, table)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{table}] 
A table containing the relative abundances of each taxa

\item[\code{year\_id}] 
Two digit code indicating the last two digits of the year of interest (05, 07, 08, 09) surrounded by quotes. Regular expressions can be used.

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns an OTU table containing only samples from the specified year
\end{Value}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(otu_table)

seven <- year_subset("07", otu_table)

# Select two years at once
two_years <- year_subset("07|08", otu_table)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{zscore}{Z-score normalize relative abundance data}{zscore}
\keyword{data\_processing}{zscore}
%
\begin{Description}\relax
Normalizes taxa abundances in a table of relative abundance data using the z-score method.
((Abundance of one OTU in one sample) - (mean abundance for that OTU ))/(standard deviation of that OTU)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
zscore(table)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{table}] 
A table of relative abundance data with taxa in rows and samples in columns

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Returns a table with relative abundance data replaced by z-scores
\end{Value}
%
\begin{Note}\relax
There is debate on whether this method of normalization is valid for microbial communities, as their abundance distrubtions tend to be heavily skewed.
I found it useful for plotting heatmaps and for input into network analysis.
\end{Note}
%
\begin{Author}\relax
Alexandra Linz <amlinz16@gmail.com>
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
data(otu_table)

# Create a small table for z-score normalization
example <- year_subset("05", otu_table)
example <- bog_subset("TBE", example)

# Remove OTUs that are not present in this subset
example <- example[which(rowSums(example) > 0), ]

z_otu_table <- zscore(example)
\end{ExampleCode}
\end{Examples}
\printindex{}
\end{document}
