---
title: "NTL-MO_figure_list_take2"
author: "Alex Linz"
date: "Sunday, August 07, 2016"
output: html_document
---

### Why the changes?

After reading our reviews from mSystems, I decided to look further into seasonal trends. My goal was to identify OTUs in the hypolimnion that change seasonally (specifically ones that are shared with Mary Lake) and try to correlate them to environmental variables collected by LTER (Long-Term Ecological Research).

```{r, setup, echo = F, message = F, warning = F}
# Set up the environment

library(OTUtable)
library(ggplot2)
library(reshape2)
library(vegan)
library(phyloseq)
library(ape)
library(indicspecies)
library(grid)
library(dplyr)

data(otu_table)
data(taxonomy)
data(metadata)

```

The key figure of the submitted paper draft was this:

```{r, sim2mary, echo = F}
# Figure 5
# Calculate Bray-Curtis Similarity for Trout Bog vs Mary Lake hypolimnia, 2007

# Create a representative sample for Mary Lake for each year and test its similarity in PCoA

MAHtable <- bog_subset("MAH", otu_table)
MAHtable05 <- year_subset("05", MAHtable)
MAHtable07 <- year_subset("07", MAHtable)
MAHtable08 <- year_subset("08", MAHtable)
MAHtable09 <- year_subset("09", MAHtable)

otu_table2 <- otu_table
otu_table2$MAH01REP05 <- rowSums(MAHtable05)/dim(MAHtable05)[2]
otu_table2$MAH01REP07 <- rowSums(MAHtable07)/dim(MAHtable07)[2]
otu_table2$MAH01REP08 <- rowSums(MAHtable08)/dim(MAHtable08)[2]
otu_table2$MAH01REP09 <- rowSums(MAHtable09)/dim(MAHtable09)[2]

seqs <- read.dna("C:/Users/Alex/Desktop/North_Temperate_Lakes-Microbial_Observatory/Data/16S_data/bog_repseqs_07Jul15.fasta", format = "fasta")
d <- dist.dna(seqs, model = "raw")
bogtree <- nj(d)

OTU2 <- otu_table(as.matrix(otu_table2), taxa_are_rows = T)
TAX <- tax_table(as.matrix(taxonomy))
sampledata2 <- sample_data(data.frame(Bog = substr(colnames(otu_table2), start = 1, stop = 2), Layer = substr(colnames(otu_table2), start = 3, stop = 3), Year = substr(colnames(otu_table2), start = 9, stop = 10), row.names = colnames(otu_table2), stringsAsfactors = F)) 
alldata_reps <- phyloseq(OTU2, TAX, sampledata2, bogtree)

TBH07 <- prune_samples(sampledata2$Year == "07" & sampledata2$Bog == "TB" & sampledata2$Layer == "H" | sampledata2$Year == "07" & substr(sample_names(sampledata2), start = 6, stop = 8) == "REP", alldata_reps)
TBH07_lake <- factor(substr(sample_names(TBH07), start = 1, stop = 2), levels = c("TB", "MA"))

x <- UniFrac(TBH07, weighted = T, normalize = T)
sim <- 1 - as.matrix(x)[1:length(TBH07_lake)-1, length(TBH07_lake)]
TBH07_date <- extract_date(names(sim))
# plot(TBH07_date[order(TBH07_date)], sim[order(TBH07_date)], type = "l")

plot.trend <- data.frame(TBH07_date, sim)
colnames(plot.trend) <- c("Date", "Distance")

TBHmat <- make_temp_matrix("TBH.....07", metadata)
TBHmat <- melt(TBHmat)
colnames(TBHmat) <- c("Depth", "Date", "Temperature")
TBHmat$Date <- as.Date(TBHmat$Date, format = "%Y-%m-%d")
TBHmat$Depth <- -TBHmat$Depth / 39 + 0.89

# Need to close polygons - add 0 or max values at top and bottom of graph
TBHmat <- TBHmat[which(is.na(TBHmat$Temperature) == F), ]
# For each date, add a hidden value outside of the plotting range
add <- TBHmat[which(TBHmat$Depth == 0.89), ]
add$Depth <- rep(-1, length(add$Depth))
add$Temperature <- rep(4, length(add$Temperature))
add2 <- add
add2$Depth <- rep(3, length(add2$Depth))
add2$Temperature <- rep(28, length(add2$Temperature))
TBHmat2 <- rbind(TBHmat, add, add2)


ggplot() + stat_contour(data = TBHmat2, aes(y = Depth, x = Date, z = Temperature, fill = ..level..), geom = "polygon") + scale_fill_gradientn(colours = c("dodgerblue", "cyan", "green", "yellow", "red"), "Temp", limits = c(4, 28)) + geom_line(data = plot.trend, aes(x = Date, y = Distance), size = 1.5) + labs(y = "1 - UniFrac Distance", x = NULL) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_rect(fill = "dodgerblue3"), axis.line = element_line(colour = "black"), axis.ticks = element_line(colour="black")) + theme(axis.text.x = element_text(hjust = 0.5, size = 12, colour = "black"), axis.title.x = element_text(size = 15, vjust = 0.2), axis.title.y = element_text(size = 12, vjust = 1.6), axis.text.y = element_text(colour = "black", size = 10)) + coord_cartesian(xlim = extract_date(c("TBH20Jun07", "TBH11Nov07")), ylim = c(0.75, 0.88))

```

My interpretation of this figure was that the community of the Trout Bog hypolimnion was becoming more similar to the community of the Mary Lake hypolimnion over each season, but was interrupted by fall mixing every year. I proposed TEA depletion as a mechanism for this trend, but since we don't have enough knowledge to link TEA use to 16S for most of the observed OTUs, I could not prove this. This was a major weakness of the paper.

After the reviews, I wanted to look into this trend more in depth. Specifically, I wanted to know which OTUs were gained over time. I assumed that as the number of OTUs shared between Trout Bog and Mary Lake increased, their abundances would also increase.

```{r, sim2mary_abundance, echo = F}
clade_table <- combine_otus("Clade", otu_table, taxonomy)

MAH <- bog_subset("MAH", clade_table)
TBH <- bog_subset("TBH", clade_table)

MAH07 <- year_subset("07", MAH)
TBH07 <- year_subset("07", TBH)

#For each sample, which clades are also found in that year of Mary Lake? How many reads are from those clades?

MAH07totals <- rowSums(MAH07)

counts <- c()
for(i in 1:dim(TBH07)[2]){
  hits <- which(TBH07[,i] > 0 & MAH07totals > 0)
  counts[i] <- sum(TBH07[hits, i])
}

TBH07date <- extract_date(colnames(TBH07))
plot(TBH07date, counts, main = "Total Read Abundance of Shared Clades")
text(x = extract_date(c("TBH15Mar07")), y = 2420, labels = paste("r^2 =", round(cor(as.numeric(TBH07date), counts), digits = 2)))

```

The plot above shows that the actual read abundance of shared taxa does not increase over time, even though the number of shared taxa does increase. I think the observed trend of increased similarity between these two locations is actually driven by biodiversity. Meromictic hypolimnia are more diverse than the other hypolimnia, so as Trout Bog gains new taxa over a season, there's a high chance that those taxa are already present in Mary Lake. This is just another way of looking at the relationship between biodiversity and mixing. This may also be driving the observed trends in the connectivity analysis.

### So what now?

I'm not confident in presenting this figure as the main message of the paper. Instead, I'd like to expand the analysis to all seasonal trends and impacts of mixing, including general ones like biodiversity. I already started doing this to support part of another project based on our metagenomic data from Trout Bog, so many of the methods I'm using have already been developed. 

The main message I'd like to get across is that epilimnia vary with seasonal factors, while hypolimnia vary with annual factors. The advantages of this approach are:
- Allows use of the entire dataset, not just hypolimnion data
- Incorporates and expands on the relationship of the hypolimnion community to mixing
- Incorporates environmental metadata better (one of the main critques of the reviewers)
- Is a more streamlined and approachable story

### Figure List, Take 2

Table 1. North Temperate Lakes - Microbial Observatory Study Sites. Lake locations, abbreviations, and physical and chemical characteristics (epilimnion/hypolimnion) are presented below. (I will make a nicer version of this in Word).

```{r, laketable, echo = F}
Lakes <- c("Forestry Bog", "Crystal Bog", "North Sparkling Bog", "West Sparkling Bog", "Trout Bog", "South Sparkling Bog", "Hell's Kitchen", "Mary Lake")
IDs <- c("FB", "CB", "NS", "WS", "TB", "SS", "HK", "MA")
Depth <- c(2, 2.5, 4.5, 4.6, 7, 8, 19.3, 21.5)
SurfaceArea <- c(1300, 5600, 4700, 11900, 10100, 4400, 30000, 12000)
Regime <- c("Polymictic", "Polymictic", "Dimictic", "Dimictic", "Dimictic", "Dimictic", "Meromictic", "Meromictic")
GPS <- c("46.047776, -89.651248", "46.007639, -89.606341", "46.004819, -89.705214", "46.004633, -89.709082", "46.041140, -89.686352", "46.041140, -89.709082", "46.186674, -89.702510", "46.250764, -89.900419")
Years <- c("2007", "2005, 2007, 2009", "2007, 2008, 2009", "2007", "2005, 2007, 2008, 2009", "2007, 2008, 2009", "2007", "2005, 2007, 2008, 2009")
DIC <- c("NA", "0.69/1.72", "1.12/2.31", "NA", "1.73/4.47", "1.97/6.42", "NA", "5.54/12.38")
DOC <- c("NA", "15.47/13.6", "10.05/10.40", "NA", "19.87/20.58", "12.40/21.92", "NA", "20.63/67.10")
TN <- c("NA", "620.57/846.00", "629.09/809.45", "NA", "737.71/1121.00", "813.88/1498", "NA", "1332.57/3652.38")
TP <- c("NA", "30.00/38.86", "78.00/135.45", "NA", "50.57/53.25", "48.63/69.14", "NA", "78.00/303.50")
TDN <- c("NA", "1290.19/490.13", "442.39/586.56", "NA", "582.5/820.21", "451.63/1179.21", "NA", "1024.5/3220.14")
TDP <- c("NA", "84.25/14.88", "70.22/22.67", "NA", "34.5/31.57", "16.25/18.29", "NA", "71.13/228")

table1 <- data.frame(Lakes, IDs, Depth, SurfaceArea, Regime, GPS, Years, DIC, DOC, TN, TP, TDN, TDP)
print(table1)

```

```{r, epi_hypo_pcoas, echo = F}
OTU <- otu_table(as.matrix(otu_table), taxa_are_rows = T)
TAX <- tax_table(as.matrix(taxonomy))

sampledata <- sample_data(data.frame(Bog = substr(colnames(otu_table), start = 1, stop = 2), Layer = substr(colnames(otu_table), start = 3, stop = 3), Year = substr(colnames(otu_table), start = 9, stop = 10), row.names = colnames(otu_table), stringsAsfactors = F))                                        

alldata <- phyloseq(OTU, TAX, sampledata, bogtree)

epi <-prune_samples(sampledata$Year == "07" & sampledata$Layer == "E", alldata)
epi_lakes <- factor(substr(sample_names(epi), start = 1, stop = 2), levels = c("FB", "CB", "WS", "NS", "TB", "SS", "HK", "MA"))
epi_regime <- rep(NA, length(epi_lakes))
epi_regime[which(epi_lakes == "FB" | epi_lakes == "CB" | epi_lakes == "WS")] <- "polymictic"
epi_regime[which(epi_lakes == "NS" | epi_lakes == "TB" | epi_lakes == "SS")] <- "dimictic"
epi_regime[which(epi_lakes == "HK" | epi_lakes == "MA")] <- "meromictic"
epi_regime <- factor(epi_regime, levels = c("polymictic", "dimictic", "meromictic"))

x <- UniFrac(epi, weighted = T, normalize = T)
pcoa <- betadisper(x, epi_lakes)
scores <- scores(pcoa)

plot.pcoa <- data.frame(scores$sites, epi_lakes, epi_regime)
colnames(plot.pcoa) <- c("PCoA1", "PCoA2", "Lake", "Regime")

axis1 <- round(pcoa$eig[1]/sum(pcoa$eig), digits = 2)
axis2 <- round(pcoa$eig[2]/sum(pcoa$eig), digits = 2)

ggplot(data=plot.pcoa, aes(x = PCoA1, y = PCoA2, color = Lake, shape = Regime)) + geom_point() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"), axis.ticks = element_line(colour = "black")) + theme(axis.text.x = element_text(hjust = 0.5, size = 10, colour = "black"), axis.text.y = element_text(size = 10, color = "black"), axis.title = element_text(size = 10, hjust = 0.5, vjust = 0.1), panel.border = element_rect(colour = "black", fill=NA, size=1)) + labs(title = "Epilimnia", x = paste("PCoA1 (", axis1, ")", sep = ""), y = paste("PCoA2 (", axis2, ")")) + scale_color_brewer(palette = "Set3")


adonis(x ~ Bog, as(sample_data(epi), "data.frame"))
# r^2 = 0.36625, p = 0.001
# Calculate PERMADISP by mixing regime
adonis(x ~ epi_regime, as(cbind(sample_data(epi), epi_regime), "data.frame"))
# r^2 = 0.19818, p = 0.001

# Repeat with hypolimnion
hypo <-prune_samples(sampledata$Year == "07" & sampledata$Layer == "H", alldata)
hypo_lakes <- factor(substr(sample_names(hypo), start = 1, stop = 2), levels = c("FB", "CB", "WS", "NS", "TB", "SS", "HK", "MA"))
hypo_regime <- rep(NA, length(hypo_lakes))
hypo_regime[which(hypo_lakes == "FB" | hypo_lakes == "CB" | hypo_lakes == "WS")] <- "polymictic"
hypo_regime[which(hypo_lakes == "NS" | hypo_lakes == "TB" | hypo_lakes == "SS")] <- "dimictic"
hypo_regime[which(hypo_lakes == "HK" | hypo_lakes == "MA")] <- "meromictic"
hypo_regime <- factor(hypo_regime, levels = c("polymictic", "dimictic", "meromictic"))

x <- UniFrac(hypo, weighted = T, normalize = T)
pcoa <- betadisper(x, hypo_lakes)
scores <- scores(pcoa)

plot.pcoa <- data.frame(scores$sites, hypo_lakes, hypo_regime)
colnames(plot.pcoa) <- c("PCoA1", "PCoA2", "Lake", "Regime")

axis1 <- round(pcoa$eig[1]/sum(pcoa$eig), digits = 2)
axis2 <- round(pcoa$eig[2]/sum(pcoa$eig), digits = 2)


ggplot(data=plot.pcoa, aes(x = PCoA1, y = PCoA2, color = Lake, shape = Regime)) + geom_point() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"), axis.ticks = element_line(colour = "black")) + theme(axis.text.x = element_text(hjust = 0.5, size = 10, colour = "black"), axis.text.y = element_text(size = 10, color = "black"), axis.title = element_text(size = 10, hjust = 0.5, vjust = 0.1), panel.border = element_rect(colour = "black", fill=NA, size=1)) + labs(title = "hypolimnia", x = paste("PCoA1 (", axis1, ")", sep = ""), y = paste("PCoA2 (", axis2, ")")) + scale_color_brewer(palette = "Set3")


adonis(x ~ Bog, as(sample_data(hypo), "data.frame"))
# r^2 = 0.49795, p = 0.001
# Calculate PERMADISP by mixing regime
adonis(x ~ hypo_regime, as(cbind(sample_data(hypo), hypo_regime), "data.frame"))
# r^2 = 0.34851, p = 0.001
```

Figure 1. Community differences by lake and mixing regime. Principle coordinates analysis was performed on epilimnion and hypolimnion samples using a weighted UniFrac distance matrix. A permutational MANOVA test was used to determine significant clustering by lake and mixing regime. Significant clustering by both characteristics was found in both layers; however, correlation coefficients were higher in hypolimnia than in epilimnia. *This suggests that epilimnion communities are more similar between lakes than hypolimnion communities.*

Table 2. Indicator Taxa by layer and mixing regime. Indicator analysis was run on datasets separate by layer, with mixing dates removed, on taxa grouped by OTU, clade, and lineage. Differences between lakes indicate potential differences in biogeochemical cycling; for example, indicator taxa from Desulfobulbaceae, Desulfobacterales, and Syntrophobacterales suggest that sulfide reduction is taking place in dimictic and meromictic hypolimnia, but not in polymictic hypolimnia. Similar taxa are observed between epilimnia and hypolimnia from the same mixing regimes, suggesting interaction between the two layers. *Differences in layers and mixing regimes are reflected in the habitat preferences of bacterial taxa.*


```{r, indic_by_layer_mixing, echo = F}
# Make a table that has OTUs grouped at every taxonomic level possible. This is so that groups at different levels will compete in the analyis.
# For example, I want to know if clade acI-B is a better or worse indicator than its phylum, Actinobacteria, so I run the analysis on both levels at once.

# Rename the OTUs with their full taxonomic assignment
named_otu_table <- otu_table
fullnames <- c()
for(i in 1:dim(taxonomy)[1]){
  fullnames[i] <- paste(taxonomy[i, ], collapse = ";")
}
fullnames <- make.unique(fullnames)
rownames(named_otu_table) <- fullnames

# Re-run clade and phylum tables so that names are no longer shortened.
clade_table <- combine_otus("Clade", otu_table, taxonomy)
phylum_table <- combine_otus("Phylum", otu_table, taxonomy)
lineage_table <- combine_otus("Lineage", otu_table, taxonomy)
order_table <- combine_otus("Order", otu_table, taxonomy)
class_table <- combine_otus("Class", otu_table, taxonomy)

# Combine the tables at all taxonomic levels into one giant table
full_table <- rbind(named_otu_table, clade_table, lineage_table)

# Remove unclassified groups - I'm not interested in these for this analysis
classified <- grep("unclassified", rownames(full_table))
classified1 <- grep("__$", rownames(full_table))
parsed_table <- full_table[-c(classified, classified1),]

# Identify mixing dates
metadata$Sample_Name <- as.character(metadata$Sample_Name)
metalakes <- substr(metadata$Sample_Name, start=1, stop=3)
metayears <- substr(metadata$Sample_Name, start=9, stop=10)
metaTBH <- metadata[which(metalakes == "TBH"), c(1,2,4)]
metaTBH <- dcast(metaTBH, Sample_Name~Depth, fun.aggregate=mean)
TBHmixes <- metaTBH$Sample_Name[which(metaTBH$"0.5" - metaTBH$"7" < 1)]

metaNSH <- metadata[which(metalakes == "NSH"), c(1,2,4)]
metaNSH <- dcast(metaNSH, Sample_Name~Depth, fun.aggregate=mean)
NSHmixes <- metaNSH$Sample_Name[which(metaNSH$"0.5" - metaNSH$"4" < 1)]

metaSSH <- metadata[which(metalakes == "SSH"), c(1,2,4)]
metaSSH <- dcast(metaSSH, Sample_Name~Depth, fun.aggregate=mean)
SSHmixes <- metaSSH$Sample_Name[which(metaSSH$"0.5" - metaSSH$"8" < 1)]

metaCBH <- metadata[which(metalakes == "CBH"), c(1,2,4)]
metaCBH <- dcast(metaCBH, Sample_Name~Depth, fun.aggregate=mean)
CBHmixes <- metaCBH$Sample_Name[which(metaCBH$"0.5" - metaCBH$"2" < 1)]

metaWSH <- metadata[which(metalakes == "WSH"), c(1,2,4)]
metaWSH <- dcast(metaWSH, Sample_Name~Depth, fun.aggregate=mean)
WSHmixes <- metaWSH$Sample_Name[which(metaWSH$"0.5" - metaWSH$"4" < 1)]

metaFBH <- metadata[which(metalakes == "FBH"), c(1,2,4)]
metaFBH <- dcast(metaFBH, Sample_Name~Depth, fun.aggregate=mean)
FBHmixes <- metaFBH$Sample_Name[which(metaFBH$"0.5" - metaFBH$"1.5" < 1)]

mixes <- c(CBHmixes, FBHmixes, WSHmixes, TBHmixes, NSHmixes, SSHmixes)

# Make separate datasets for each mixing regime
# Note: not using "NS." because there are some "NSU" (layer unknown) samples from that lake
epi_table <- bog_subset("..E", parsed_table)
mixing_dates <- match(mixes, substr(colnames(epi_table), start=1, stop=10))
remove <- mixing_dates[which(is.na(mixing_dates) == F)]
if(length(remove) > 0){
  epi_table <- epi_table[,-remove]
}


# Keep only groups with abundances in the top quantile (75th or higher)
threshold <- quantile(rowSums(epi_table))[4]
epi_table <- epi_table[which(rowSums(epi_table) >= threshold),]

# Format table for input into indicspecies analysis
epi_table <- t(epi_table)
epi_table <- as.data.frame(epi_table)
# Group by layer identifier
sampleids <- rownames(epi_table)
lake <- substr(sampleids, start = 1, stop = 2)
layerid <- c("FB", "CB", "NS", "WS", "TB", "SS", "HK", "MA")

regime <- c()
regime[which(lake == "FB" | lake == "CB" | lake == "WS")] <- 1
regime[which(lake == "NS" | lake == "TB" | lake == "SS")] <- 2
regime[which(lake == "MA" | lake == "HK")] <- 3

# Run multipatt(), specifing the the index of choice is group-normalized correlation
epi_indic <- multipatt(x = epi_table, cluster = regime, func = "r.g", control = how(nperm = 9999))

#List the top 10 strongest indicators for both epi and hypo
epi_results <- epi_indic$sign
epi_results <- epi_results[which(epi_results$p.value < 0.05),]

poly_epi_results <- epi_results[which(epi_results$index == 1), ]
poly_epi_results <- poly_epi_results[order(poly_epi_results$stat, decreasing = T),]
#Choose the top 10 manually, so that if lineage is a better indicator than clade, only the highest level is reported
poly_epi_indicators <- rownames(poly_epi_results)[c(1, 2, 3, 5, 6, 10, 13, 14, 17)]
#Only 9, so adding a blank at the end of the vector
poly_epi_indicators[10] <- " "

di_epi_results <- epi_results[which(epi_results$index == 2), ]
di_epi_results <- di_epi_results[order(di_epi_results$stat, decreasing = T),]
di_epi_indicators <- rownames(di_epi_results)[c(1, 2, 4, 5, 6, 7, 8, 9, 10, 11)]

mero_epi_results <- epi_results[which(epi_results$index == 3), ]
mero_epi_results <- mero_epi_results[order(mero_epi_results$stat, decreasing = T),]
mero_epi_indicators <- rownames(mero_epi_results)[c(1, 3, 4, 5, 6, 8, 9, 11, 12, 13)]

# Repeat with hypolimnion
hypo_table <- bog_subset("..H", parsed_table)
mixing_dates <- match(mixes, substr(colnames(hypo_table), start=1, stop=10))
remove <- mixing_dates[which(is.na(mixing_dates) == F)]
if(length(remove) > 0){
  hypo_table <- hypo_table[,-remove]
}


# Keep only groups with abundances in the top quantile (75th or higher)
threshold <- quantile(rowSums(hypo_table))[4]
hypo_table <- hypo_table[which(rowSums(hypo_table) >= threshold),]

# Format table for input into indicspecies analysis
hypo_table <- t(hypo_table)
hypo_table <- as.data.frame(hypo_table)
# Group by layer identifier
sampleids <- rownames(hypo_table)
lake <- substr(sampleids, start = 1, stop = 2)
layerid <- c("FB", "CB", "NS", "WS", "TB", "SS", "HK", "MA")

regime <- c()
regime[which(lake == "FB" | lake == "CB" | lake == "WS")] <- 1
regime[which(lake == "NS" | lake == "TB" | lake == "SS")] <- 2
regime[which(lake == "MA" | lake == "HK")] <- 3

# Run multipatt(), specifing the the index of choice is group-normalized correlation
hypo_indic <- multipatt(x = hypo_table, cluster = regime, func = "r.g", control = how(nperm = 9999))

#List the top 10 strongest indicators for both hypo and hypo
hypo_results <- hypo_indic$sign
hypo_results <- hypo_results[which(hypo_results$p.value < 0.05),]

poly_hypo_results <- hypo_results[which(hypo_results$index == 1), ]
poly_hypo_results <- poly_hypo_results[order(poly_hypo_results$stat, decreasing = T),]
#Choose the top 10 manually, so that if lineage is a better indicator than clade, only the highest level is reported
poly_hypo_indicators <- rownames(poly_hypo_results)[c(1, 4, 6, 11, 15, 17, 18, 19, 21, 23)]

di_hypo_results <- hypo_results[which(hypo_results$index == 2), ]
di_hypo_results <- di_hypo_results[order(di_hypo_results$stat, decreasing = T),]
di_hypo_indicators <- rownames(di_hypo_results)[c(1, 2, 3, 4, 5, 6, 8, 9, 11, 12)]

mero_hypo_results <- hypo_results[which(hypo_results$index == 3), ]
mero_hypo_results <- mero_hypo_results[order(mero_hypo_results$stat, decreasing = T),]
mero_hypo_indicators <- rownames(mero_hypo_results)[c(1, 2, 3, 5, 6, 8, 9, 10, 11, 12)]

# Make tables of all these results

table2a <- data.frame(poly_epi_indicators, di_epi_indicators, mero_epi_indicators)
table2b <- data.frame(poly_hypo_indicators, di_hypo_indicators, mero_hypo_indicators)

print(table2a)
print(table2b)
```

```{r, pcoas_by_lake, echo = F}

# First make a PCoA of TBE
TBE <-prune_samples(sampledata$Bog == "TB" & sampledata$Layer == "E", alldata)
TBE <- prune_taxa(taxa_sums(TBE) > 100, TBE)
TBEmonth <- factor(substr(rownames(TBE@sam_data), start = 6, stop = 8), levels = c("JAN", "Feb", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV"))

x <- UniFrac(TBE, weighted = T, normalize = T)
pcoa <- betadisper(x, TBE@sam_data$Year)
scores <- scores(pcoa)

plot.pcoa <- data.frame(scores$sites, TBE@sam_data$Year, TBEmonth)
colnames(plot.pcoa) <- c("PCoA1", "PCoA2", "Year", "Month")

axis1 <- round(pcoa$eig[1]/sum(pcoa$eig), digits = 2)
axis2 <- round(pcoa$eig[2]/sum(pcoa$eig), digits = 2)

ggplot(data=plot.pcoa, aes(x = PCoA1, y = PCoA2, color = Month, shape = Year)) + geom_point(size = 3) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"), axis.ticks = element_line(colour = "black")) + theme(axis.text.x = element_text(hjust = 0.5, size = 10, colour = "black"), axis.text.y = element_text(size = 10, color = "black"), axis.title = element_text(size = 10, hjust = 0.5, vjust = 0.1), panel.border = element_rect(colour = "black", fill=NA, size=1)) + labs(title = "Trout Bog Epilimnion", x = paste("PCoA1 (", axis1, ")", sep = ""), y = paste("PCoA2 (", axis2, ")")) + scale_color_brewer(palette = "Set3")

adonis(x ~ Year, as(sample_data(TBE), "data.frame"))
# r^2 = 0.12761, p = 0.001
# Calculate PERMADISP by mixing regime
adonis(x ~ TBEmonth, as(cbind(sample_data(TBE), TBEmonth), "data.frame"))
# r^2 = 0.52258, p = 0.001

# Month is a better predictor than year for TBE. Time since ice-off would probably be even better.

# What about TBH?

TBH <-prune_samples(sampledata$Bog == "TB" & sampledata$Layer == "H", alldata)
TBH <- prune_taxa(taxa_sums(TBH) > 100, TBH)
TBHmonth <- factor(substr(rownames(TBH@sam_data), start = 6, stop = 8), levels = c("JAN", "Feb", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV"))

x <- UniFrac(TBH, weighted = T, normalize = T)
pcoa <- betadisper(x, TBH@sam_data$Year)
scores <- scores(pcoa)

plot.pcoa <- data.frame(scores$sites, TBH@sam_data$Year, TBHmonth)
colnames(plot.pcoa) <- c("PCoA1", "PCoA2", "Year", "Month")

axis1 <- round(pcoa$eig[1]/sum(pcoa$eig), digits = 2)
axis2 <- round(pcoa$eig[2]/sum(pcoa$eig), digits = 2)

ggplot(data=plot.pcoa, aes(x = PCoA1, y = PCoA2, color = Month, shape = Year)) + geom_point(size = 3) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"))  + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"), axis.ticks = element_line(colour = "black")) + theme(axis.text.x = element_text(hjust = 0.5, size = 10, colour = "black"), axis.text.y = element_text(size = 10, color = "black"), axis.title = element_text(size = 10, hjust = 0.5, vjust = 0.1), panel.border = element_rect(colour = "black", fill=NA, size=1)) + labs(title = "Trout Bog Hypolimnion", x = paste("PCoA1 (", axis1, ")", sep = ""), y = paste("PCoA2 (", axis2, ")")) + scale_color_brewer(palette = "Set3")

adonis(x ~ Year, as(sample_data(TBH), "data.frame"))
# r^2 = 0.33359, p = 0.001
# Calculate PERMADISP by mixing regime
adonis(x ~ TBHmonth, as(cbind(sample_data(TBH), TBHmonth), "data.frame"))
# r^2 = 0.35017, p = 0.001

# Year is more of a driver in hypolimnia, nearly equivalent with month.
```

Figure 2. PCoA of the Trout Bog epilimnion and hypolimnion. This analysis shows which samples are most similar to each other. In the epilimnion, month is a strong predictor of sample similarity, regardless of year. Year is a much stronger determinant in the hypolimnion. In both layers, samples taken after the fall mixing event (late November, January and February samples) are most similar, regardless of year.



```{r, adonis_time, echo = F}

#Write a function to return the R2 values of adonis, given the lake layer I want to test

adonis_pipeline <- function(lake, layer){
  dataset <- prune_samples(sampledata$Bog == lake & sampledata$Layer == layer, alldata)
  dataset <- prune_taxa(taxa_sums(dataset) > length(unique(dataset@sam_data$Year))*25, dataset)
  month <- factor(substr(rownames(dataset@sam_data), start = 6, stop = 8), levels = c("JAN", "Feb", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV"))
  if(lake == "NS" & layer == "H"){
    month[17] <- "JUL"
  }

  x <- UniFrac(dataset, weighted = T, normalize = T)

  yearfactor <- adonis(x ~ Year, as(sample_data(dataset), "data.frame"))
  monthfactor <- adonis(x ~ month, as(cbind(sample_data(dataset), month), "data.frame"))
  results <- c(yearfactor$aov.tab$R2[1], monthfactor$aov.tab$R2[1])
  names(results) <- c("Year", "Month")
  return(results)
}

# Run this on lakes and layers with multiple years of data

CBE_factors <- adonis_pipeline("CB", "E")
CBH_factors <- adonis_pipeline("CB", "H")
NSE_factors <- adonis_pipeline("NS", "E")
NSH_factors <- adonis_pipeline("NS", "H")
TBE_factors <- adonis_pipeline("TB", "E")
TBH_factors <- adonis_pipeline("TB", "H")
SSE_factors <- adonis_pipeline("SS", "E")
SSH_factors <- adonis_pipeline("SS", "H")
MAE_factors <- adonis_pipeline("MA", "E")
MAH_factors <- adonis_pipeline("MA", "H")

all_factors <- rbind(CBE_factors, CBH_factors, NSE_factors, NSH_factors, TBE_factors, TBH_factors, SSE_factors, SSH_factors, MAE_factors, MAH_factors)

all_factors <- cbind(all_factors, substr(rownames(all_factors), start = 1, stop = 3))
colnames(all_factors)[3] <- "Key"
all_factors <- data.frame(all_factors)
all_factors2 <- melt(all_factors, id = "Key")

ggplot(data = all_factors2, aes(x = Key, y = value, fill = variable)) + geom_bar(stat = "identity", position = "dodge") + theme_bw()
```


Figure 3. Contribution of year and month to community composition. A permutational MANOVA was performed on a weighted UniFrac distance matrix constructed from samples split by lake and layer. Month describes more variation in community composition than year in nearly every subset of data, but year describes more variation in hypolimnia than in epilimnia.


```{r, acI_by_lake, echo = F}
acI <- grab_group("^acI$", "Lineage", otu_table, taxonomy)
acI$Taxonomy <- colsplit(rownames(acI), ";", names = colnames(taxonomy))$Tribe
acI <- acI[grep("unclassified", acI$Taxonomy, invert = T),]
acI <- acI[which(rowSums(acI[1:1387]) > 5000), ]

acI <- melt(acI)
acI$Lake <- factor(substr(acI$variable, start = 1, stop = 2), levels = c("CB", "FB", "NS", "WS", "TB", "SS", "HK", "MA"))
acI$Layer <- substr(acI$variable, start = 3, stop = 3)
acI$Date <- extract_date(acI$variable)
acI$Julian <- as.POSIXct(format(acI$Date, format = "%j"), format = "%j")
acI <- acI[which(acI$Layer != "U"),]

acI_by_lake <- group_by(acI, Taxonomy, Lake, Layer)
acI_by_lake <- summarize(acI_by_lake, abundance = mean(value))

acI_epi <- filter(acI_by_lake, Layer == "E")
ggplot(acI_epi, aes(x = Lake, y = Taxonomy, fill = abundance)) + geom_tile() + labs(title = "Mean Epilimnion Abundance") + theme_bw() + scale_fill_gradient2(low = "white", mid = "purple", high = "darkblue", midpoint = 175)

acI_hypo <- filter(acI_by_lake, Layer == "H")
ggplot(acI_hypo, aes(x = Lake, y = Taxonomy, fill = abundance)) + geom_tile() + labs(title = "Mean Hypolimnion Abundance") + theme_bw() + scale_fill_gradient2(low = "white", mid = "purple", high = "darkblue", midpoint = 175)
```

Figure 4. Freshwater lineage acI tribes over time. Only tribes with total read abundance of greater than 5000 in the the entire dataset are shown. acI-B tribes dominate, and acI prefers epilimnia to hypolimnia.

```{r, acI_over_time, echo = F}
#Over time

acI_CBE <- filter(acI, Lake == "CB", Layer == "E")
ggplot(data = acI_CBE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + ylim(c(0, max(acI_CBE$value))) + labs(title = "Crystal Bog Epilimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Spectral")

acI_TBE <- filter(acI, Lake == "TB", Layer == "E")
ggplot(data = acI_TBE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("TBE15May16", "TBE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(acI_TBE$value))) + labs(title = " Trout Bog Epilimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Spectral")

acI_NSE <- filter(acI, Lake == "NS", Layer == "E")
ggplot(data = acI_NSE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("NSE15May16", "NSE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(acI_NSE$value))) + labs(title = "North Sparkling Bog Epilimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Spectral")


acI_SSE <- filter(acI, Lake == "SS", Layer == "E")
ggplot(data = acI_SSE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("SSE15May16", "SSE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(acI_SSE$value))) + labs(title = "South Sparkling Bog Epilimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Spectral")


acI_MAE <- filter(acI, Lake == "MA", Layer == "E")
ggplot(data = acI_MAE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("MAE15May16", "MAE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(acI_MAE$value))) + labs(title = "Mary Lake Epilimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Spectral")

```

Figure 5. acI tribes by lake epilimnia, with trends representated by a smoothed conditional loess mean. In most lakes, acI-B OTUs of the same tribe (often B2) are anti-correlated, suggesting competition. acI-B OTUs of different tribes (B2, B3, and B4) often show similar trends. In Mary Lake, acI-A dominates instead of acI-B, suggesting sufficiently different environmental conditions to favor acI-A over acI-B.

Other options:
Pnec is heavily dominate by PnecC
Lhab tribes all show nearly the same trends
Only 1 Luna tribe (A1)
acV dominated by one tribe (A2)
bacI has two competeing A1 tribes in two lakes, but no interactions with others
alfI has one of each tribe. none of the alfs have enough OTUs.
betIII has two A1s, but no abundant comparisons







```{r, betI_by_lake}
Geobacter <- grab_group("Geobacter", "Clade", otu_table, taxonomy)
Geobacter$Taxonomy <- colsplit(rownames(Geobacter), ";", names = colnames(taxonomy))$Tribe
Geobacter <- Geobacter[which(rowSums(Geobacter[1:1387]) > 500), ]

Geobacter <- melt(Geobacter)
Geobacter$Lake <- factor(substr(Geobacter$variable, start = 1, stop = 2), levels = c("CB", "FB", "NS", "WS", "TB", "SS", "HK", "MA"))
Geobacter$Layer <- substr(Geobacter$variable, start = 3, stop = 3)
Geobacter$Date <- extract_date(Geobacter$variable)
Geobacter$Julian <- as.POSIXct(format(Geobacter$Date, format = "%j"), format = "%j")
Geobacter <- Geobacter[which(Geobacter$Layer != "U"),]

Geobacter_by_lake <- group_by(Geobacter, Taxonomy, Lake, Layer)
Geobacter_by_lake <- summarize(Geobacter_by_lake, abundance = mean(value))

Geobacter_epi <- filter(Geobacter_by_lake, Layer == "E")
ggplot(Geobacter_epi, aes(x = Lake, y = Taxonomy, fill = abundance)) + geom_tile() + labs(title = "Mean Epilimnion Abundance") + theme_bw() + scale_fill_gradient2(low = "white", mid = "purple", high = "darkblue", midpoint = 25)

Geobacter_hypo <- filter(Geobacter_by_lake, Layer == "H")
ggplot(Geobacter_hypo, aes(x = Lake, y = Taxonomy, fill = abundance)) + geom_tile() + labs(title = "Mean Hypolimnion Abundance") + theme_bw() + scale_fill_gradient2(low = "white", mid = "purple", high = "darkblue", midpoint = 25)
```

```{r, Geobacter_over_time}
Geobacter_CBE <- filter(Geobacter, Lake == "CB", Layer == "H")
ggplot(data = Geobacter_CBE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + ylim(c(0, max(Geobacter_CBE$value))) + labs(title = "Crystal Bog Hypolimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Dark2")

Geobacter_TBE <- filter(Geobacter, Lake == "TB", Layer == "H")
ggplot(data = Geobacter_TBE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("TBE15May16", "TBE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(Geobacter_TBE$value))) + labs(title = " Trout Bog Hypolimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Dark2")

Geobacter_NSE <- filter(Geobacter, Lake == "NS", Layer == "H")
ggplot(data = Geobacter_NSE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("NSE15May16", "NSE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(Geobacter_NSE$value))) + labs(title = "North Sparkling Bog Hypolimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Dark2")


Geobacter_SSE <- filter(Geobacter, Lake == "SS", Layer == "H")
ggplot(data = Geobacter_SSE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("SSE15May16", "SSE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(Geobacter_SSE$value))) + labs(title = "South Sparkling Bog Hypolimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Dark2")


Geobacter_MAE <- filter(Geobacter, Lake == "MA", Layer == "H")
ggplot(data = Geobacter_MAE, aes(x = Julian, y = value, color = Taxonomy)) + geom_point() + geom_smooth(fill = NA, span = 0.2) + theme_bw() + xlim(as.POSIXct(format(extract_date(c("MAE15May16", "MAE20Nov16")), format = "%j"), format = "%j")) + ylim(c(0, max(Geobacter_MAE$value))) + labs(title = "Mary Lake Hypolimnion", x = "Julian Date", y = "Read Abundance") + scale_color_brewer(palette = "Dark2")

```